woocommerce/woocommerce
# Customize Store -- Show warning modal when the active theme has mods

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





** This PR doesn't implement modal logic for the `Create a new one` flow.

Closes #40328 

This PR displays a warning modal when the `Design with A.I` button is clicked, but the `Customize Your Store` task has not been completed, and the active theme has modifications.

- Added `/wp-json/wc-admin/onboarding/themes/count-active-theme-mods` endpoint to count # of mods with the active theme.
- Fetches `count-active-theme-mods` and `customize-store` task in `preIntro` state



### How to test the changes in this Pull Request:

This feature is part of the larger customize your store development and behind a feature flag, hence QA review can be deferred until the call for testing before release.



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch. 
2. Enable `customize-store` feature flag.
3. Complete the core profiler.
4. Make at least one change in `Appearance -> Editor`
5. Go to `WooCommerce -> Home` and click on `Customize your store` task
6. Click on `Design with AI` button.
7. You should see a warning modal.

![Screen Shot 2023-09-21 at 4 50 14 PM](https://github.com/woocommerce/woocommerce/assets/4723145/918bcb97-a72c-4b6f-a8d3-936e489fcfb5)




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

This PR displays a warning modal when the `Design with A.I` button is clicked, but the `Customize Your Store` task has not been completed, and the active theme has modifications.

#### Comment 

</details>
